catroid/src/androidTest/java/org/catrobat/catroid/test/common/DeviceVariableAccessorTest.java

```diff
@@ -0,0 +1,180 @@
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.catroid.test.common;
+
+import android.content.Context;
+import android.support.test.InstrumentationRegistry;
+import android.support.test.runner.AndroidJUnit4;
+
+import org.catrobat.catroid.ProjectManager;
+import org.catrobat.catroid.common.defaultprojectcreators.DefaultProjectCreator;
+import org.catrobat.catroid.content.Project;
+import org.catrobat.catroid.content.Sprite;
+import org.catrobat.catroid.exceptions.LoadingProjectException;
+import org.catrobat.catroid.formulaeditor.UserVariable;
+import org.catrobat.catroid.io.DeviceVariableAccessor;
+import org.catrobat.catroid.io.XstreamSerializer;
+import org.catrobat.catroid.io.asynctask.ProjectCopyTask;
+import org.catrobat.catroid.test.utils.TestUtils;
+import org.catrobat.catroid.ui.recyclerview.controller.SpriteController;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(AndroidJUnit4.class)
+public class DeviceVariableAccessorTest {
+
+	UserVariable userVariable;
+	UserVariable localUserVariable;
+	Sprite sprite;
+	Project project;
+	Context context;
+	ArrayList<String> cleanupList = new ArrayList<>();
+
+	@Before
+	public void setUp() throws IOException {
+		context = InstrumentationRegistry.getTargetContext();
+		project = createSimpleProject("projectX");
+		sprite = new Sprite("_sprite_");
+		project.getDefaultScene().addSprite(sprite);
+		userVariable = new UserVariable("X", 10.0);
+		localUserVariable = new UserVariable("Y", 20.0);
+
+		ProjectManager.getInstance().setCurrentProject(project);
+		sprite.addUserVariable(localUserVariable);
+		project.addUserVariable(userVariable);
+		XstreamSerializer.getInstance().saveProject(project);
+		cleanupList.add(project.getName());
+	}
+
+	@Test
+	public void saveUserVariableTest() throws IOException {
+		DeviceVariableAccessor accessor = new DeviceVariableAccessor(project.getDirectory());
+
+		Object savedVar = userVariable.getValue();
+		Object localSavedVar = localUserVariable.getValue();
+		accessor.writeVariable(userVariable);
+		accessor.writeVariable(localUserVariable);
+		userVariable.setValue("false");
+		localUserVariable.setValue("false");
+		assertTrue(accessor.readUserVariableValue(userVariable));
```

Comment:
please test load and save separately.
just save a variable and assert that the resulting file contains the right values

and another test where you take a file from test assets or create one and load variables from there and assert that the values are correct.

also split up local and global variables into separate tests